File: 917-boats-to-save-people/boats-to-save-people.py
class Solution:
    def numRescueBoats(self, people: List[int], limit: int) -> int:
        people.sort()
        st,end=0,len(people)-1
        ans=0

        while st<=end:
            if people[st]+people[end]<=limit:
                st+=1
            end-=1
            ans+=1

        return ans

File: 917-boats-to-save-people/boats-to-save-people.cpp
class Solution {
public:
    int numRescueBoats(vector<int>& people, int limit) {
        sort(people.begin(),people.end());
        int st=0,end=people.size()-1;
        int ans=0;

        while (st<=end){
            if (people[st]+people[end]<=limit)
                st+=1;
            end-=1;
            ans+=1;
        }

        return ans;
    }
};